File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeHandleMountStableMessage.scala





















// Generated on 05/03/2013 10:34:53
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeHandleMountStableMessage extends MessageDeserializer {
  val messageId = 5965

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeHandleMountStableMessage.messageId

    val actionType: Byte
    val rideId: Int

    def serialize(buf: Buffer) {
      buf.writeByte(actionType)
      buf.writeInt(rideId)
    }

  }

  protected def create(actionType: Byte, rideId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val actionType = buf.readByte()
    val rideId = buf.readInt()

    create(actionType, rideId)
  }

}

object ExchangeHandleMountStableMessage extends ExchangeHandleMountStableMessage {
  case class MessageImpl(actionType: Byte, rideId: Int) extends Message
  type Target = MessageImpl

  override def create(actionType: Byte, rideId: Int) = MessageImpl(actionType, rideId)

  def apply(actionType: Byte, rideId: Int) = create(actionType, rideId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeHandleMountStableMessage.java


// Generated on 05/08/2013 19:37:53
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeHandleMountStableMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5965;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte actionType;
    public int rideId;
    
    public ExchangeHandleMountStableMessage() { }
    
    public ExchangeHandleMountStableMessage(byte actionType, int rideId) {
        this.actionType = actionType;
        this.rideId = rideId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(actionType);
        buf.writeInt(rideId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        actionType = buf.readByte();
        rideId = buf.readInt();
        if (rideId < 0)
            throw new RuntimeException("Forbidden value on rideId = " + rideId + ", it doesn't respect the following condition : rideId < 0");
    }
    
}
